bin/lib/check_reqs.js

```diff
@@ -19,15 +19,226 @@
+ */
+function getInstalledVSVersions() {
+    // Query all keys with Install value equal to 1, then filter out
+    // those, which are not related to VS itself
+    return exec('reg query HKLM\\SOFTWARE\\Microsoft\\DevDiv\\vs\\Servicing /s /v Install /f 1 /d /e /reg:32')
+    .fail(function () { return ''; })
+    .then(function (output) {
+        return output.split('\n')
+        .reduce(function (installedVersions, line) {
+            var match = /(\d+\.\d+)\\(ultimate|professional|premium|community)/.exec(line);
+            if (match && match[1] && installedVersions.indexOf(match[1]) === -1)
+                installedVersions.push(match[1]);
+            return installedVersions;
+        }, []);
+    })
+    .then(function (installedVersions) {
+        // If there is no VS2013 installed, the we have nothing to do
+        if (installedVersions.indexOf('12.0') === -1) return installedVersions;
+
+        // special case for VS 2013. We need to check if VS2013 update 2 is installed
+        return exec('reg query "HKLM\\SOFTWARE\\Microsoft\\Updates\\Microsoft Visual Studio' +
+            ' 2013\\vsupdate_KB2829760" /v PackageVersion /reg:32')
+        .then(function (output) {
+            var updateVer = Version.fromString(/PackageVersion\s+REG_SZ\s+(.*)/i.exec(output)[1]);
+            // if update version is lover than Update2, reject the promise
+            if (VS2013_UPDATE2_RC.gte(updateVer)) return Q.reject();
+            return installedVersions;
+        })
+        .fail(function () {
+            // if we got any errors on previous steps, we're assuming that
+            // required VS update is not installed.
+            installedVersions.splice(installedVersions.indexOf('12.0'));
+            return installedVersions;
+        });
+    });
+}
+
+/**
+ * Gets list of installed Windows SDKs
+ *
+ * @return  {Version[]}  List of installed SDKs' versions
+ */
+function getInstalledWindowsSdks () {
+    var installedSdks = [];
+    return exec('reg query "HKLM\\SOFTWARE\\Microsoft\\Microsoft SDKs\\Windows" /s /v InstallationFolder /reg:32')
+    .fail(function () { return ''; })
+    .then(function (output) {
+        var re = /\\Microsoft SDKs\\Windows\\v(\d+\.\d+)\s*InstallationFolder\s+REG_SZ\s+(.*)/gim;
+        var match;
+        while ((match = re.exec(output))){
+            var sdkPath = match[2];
+            // Verify that SDKs is really installed by checking SDKManifest file at SDK root
+            if (shell.test('-e', path.join(sdkPath, 'SDKManifest.xml'))) {
+                installedSdks.push(Version.tryParse(match[1]));
+            }
+        }
+    })
+    .thenResolve(installedSdks);
+}
+
+/**
+ * Gets list of installed Windows Phone SDKs. Separately searches for 8.1 Phone
+ *   SDK and Windows 10 SDK, because the latter is needed for both Windows and
+ *   Windows Phone applications.
+ *
+ * @return  {Version[]}  List of installed Phone SDKs' versions.
+ */
+function getInstalledPhoneSdks () {
```

Comment:
No, this searches for both WP8.1 and Win10 SDKs - Win10 SDK is required wor building WP10 apps.